app/Main.hs

```diff
@@ -56,17 +58,31 @@ data IndexInfo =
     { posts :: [Post]
     } deriving (Generic, Show, FromJSON, ToJSON)
 
+-- | Data for tags
+data Tag =
+  Tag { tag :: String
+      } deriving (Generic, Show, FromJSON, ToJSON, Binary, Eq, Ord)
 -- | Data for a blog post
 data Post =
-    Post { title   :: String
-         , author  :: String
-         , content :: String
-         , url     :: String
-         , date    :: String
-         , image   :: Maybe String
+    Post { title       :: String
+         , author      :: String
+         , content     :: String
+         , url         :: String
+         , date        :: String
+         , tags        :: [Tag]
```

Comment:
This can probably just be a list of Strings to avoid the weird nesting in the blog posts 😄